WIP subscriptions: Migrate to HTML5 color picker from spectrum.

majordwarf opened the pull request:
Earlier Zulip was using `spectrum-colorpicker` which seemed to be an
abandoned project (last commit 3 years ago) and since we have dropped
support for IE11, switching to HTML5 native color picker was right choice.

The following commit removes absolute code referring to
`spectrum-colorpicker` and introduces the support for HTML5 color picker
for changing stream colors. Now user can change stream colors from the
sidebar itself or using streams settings menu using HTML5 native color
picker with an option of using presets that works in all current browsers.

Fixes #14961.

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
Changing Color From Sidebar:
![Sidebar](https://user-images.githubusercontent.com/23737560/87484400-a8838c00-c653-11ea-8f4d-7130fd886072.gif)

Changing Color From Stream Setting Menu:
![Stream Settings](https://user-images.githubusercontent.com/23737560/87484414-b0dbc700-c653-11ea-91eb-b640ecf07f7c.gif)

andersk commented on static/js/stream_color.js:
```diff
@@ -37,36 +44,24 @@ function update_historical_message_color(stream_name, color) {
-    });
-};
+function hex(x) {
+    return isNaN(x) ? "00" : hexDigits[(x - x % 16) / 16] + hexDigits[x % 16];
```
`x.toString(16).padStart(2, "0")`

andersk commented on static/styles/zulip.scss:
```diff
@@ -33,7 +33,7 @@ html {
 
 body {
-    font-family: 'Source Sans Pro', 'Helvetica Neue', Helvetica, Arial, sans-serif;
+    font-family: "Source Sans Pro", "Helvetica Neue", Helvetica, Arial, sans-serif;
```
There are a lot of formatting-only changes mixed in with this commit, can you remove them?

andersk commented on tools/linter_lib/custom_check.py:
```diff
@@ -600,6 +600,10 @@
+
+         # Color picker preset swatches are generated using HBS each loop
+         'static/templates/stream_sidebar_color_picker.hbs',
+         'static/templates/subscription_color_picker.hbs'
```
Missing trailing comma.

andersk commented on version.py:
```diff
@@ -44,4 +44,4 @@
 #   minor version bump suffices.
 
-PROVISION_VERSION = '89.1'
+PROVISION_VERSION = '90'
```
Should be `90.0`.

timabbott commented:
@majordwarf have you addressed the feedback from @andersk here?  I don't see a comment to that effect.

majordwarf commented:
@timabbott Yep, I fixed the mentioned issues he stated and reverted the following on CZO. One of the things I faced was the inclusion of `prettier` in newer version of master branch so on rebasing I started getting `prettier warning` in the JS files, should I keep formatting in the same commit?

timabbott commented:
Yes, like all of our other linters and tests, `prettier` should pass on every commit.

timabbott commented:
Closing in favor of the more current https://github.com/zulip/zulip/pull/26637 for this issue.
